### PR TITLE
Added second template argument to BlockMatrixArray representing gener…

### PR DESCRIPTION
**Pull request by maieneuro**
Added second template argument to BlockMatrixArray representing generic BlockVector (Please see thread titled "BlockMatrixArray with Trilinos" in dealii-developers mailing group)

### CODE REVIEW

File: source/lac/block_matrix_array.cc [hunk 1]
```diff
@@ -426,4 +426,7 @@ template class BlockMatrixArray<double>;
 template class BlockTrianglePrecondition<float>;
 template class BlockTrianglePrecondition<double>;
 
+template class BlockMatrixArray<double, TrilinosWrappers::MPI::BlockVector>;
+template class BlockMatrixArray<float, TrilinosWrappers::MPI::BlockVector>;
```
Comment (bangerth) [inline source/lac/block_matrix_array.cc]: You'll need to guard this with something like `#ifdef DEAL_II_HAVE_TRILINOS` (compare the syntax in the other files in that directory) to ensure it also compiles for people without trilinos. You probably also want to instantiate it for the PETScWrappers block vectors.

**Comment by bangerth** [comment]
Great, thanks very much for the patch! Can you address the one comment I saw?

I'm afraid I will also have to ask you to fix the indentation. To this end, please install astyle 2.04 (this exact version, nothing newer nor older) and then run the `./contrib/utilities/indent` script that simply indents the entire code base according to the default style. I know it's a bit of a tedious step the first time around, but it helps us keep the code base in shape and easy to read.

Thanks again!

**Comment by maieneuro** [comment]
Thanks for the feedback. I've addressed the comment and run the indentation script.

**Comment by bangerth** [comment]
Thanks a lot! We hope for more patches from Africa!

**Pull request merged**